### PR TITLE
fix20738: won't error out with EOF

### PR DESCRIPTION
**Pull request by shuoliugit**

### CODE REVIEW

**Review by zhiqiangjin — DISMISSED** [review]

File: aviatrix/resource_aviatrix_controller_cert_domain_config.go [hunk 1]
```diff
@@ -37,7 +37,18 @@ func resourceAviatrixControllerCertDomainConfigCreate(ctx context.Context, d *sc
 
 	err := client.SetCertDomain(ctx, certDomain)
 	if err != nil {
-		return diag.Errorf("could not set cert domain: %v", err)
+		if strings.Contains(err.Error(), "EOF") {
+			certDomainConfig, err := client.GetCertDomain(ctx)
+			if err != nil {
+				return diag.Errorf("could not get cert domain: %v", err)
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_cert_domain_config.go]: Can we use a more unique error string here?

File: aviatrix/resource_aviatrix_controller_cert_domain_config.go [hunk 2]
```diff
@@ -37,7 +37,18 @@ func resourceAviatrixControllerCertDomainConfigCreate(ctx context.Context, d *sc
...
+			}
+			if certDomainConfig.CertDomain == certDomain {
+				d.SetId(strings.Replace(client.ControllerIP, ".", "-", -1))
+				return resourceAviatrixControllerCertDomainConfigRead(ctx, d, meta)
+			}
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_cert_domain_config.go]: What should happen if `certDomainConfig.CertDomain != certDomain`? Should we return an error?

File: aviatrix/resource_aviatrix_controller_cert_domain_config.go [hunk 3]
```diff
@@ -37,7 +37,18 @@ func resourceAviatrixControllerCertDomainConfigCreate(ctx context.Context, d *sc
...
 	if err != nil {
-		return diag.Errorf("could not set cert domain: %v", err)
+		if strings.Contains(err.Error(), "EOF") {
+			certDomainConfig, err := client.GetCertDomain(ctx)
+			if err != nil {
+				return diag.Errorf("could not get cert domain: %v", err)
+			}
+			if certDomainConfig.CertDomain == certDomain {
+				d.SetId(strings.Replace(client.ControllerIP, ".", "-", -1))
+				return resourceAviatrixControllerCertDomainConfigRead(ctx, d, meta)
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_cert_domain_config.go]: Also we should not return early if the operation is successful, just let the function return at the bottom of the function

File: aviatrix/resource_aviatrix_controller_cert_domain_config.go [hunk 4]
```diff
@@ -68,7 +79,17 @@ func resourceAviatrixControllerCertDomainConfigUpdate(ctx context.Context, d *sc
...
+				certDomainConfig, err := client.GetCertDomain(ctx)
+				if err != nil {
+					return diag.Errorf("could not get cert domain: %v", err)
+				}
+				if certDomainConfig.CertDomain == d.Get("cert_domain") {
+					return resourceAviatrixControllerCertDomainConfigRead(ctx, d, meta)
+				}
+			} else {
+				return diag.Errorf("could not update cert domain: %v", err)
+			}
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_cert_domain_config.go]: Same comments as above

File: aviatrix/resource_aviatrix_controller_cert_domain_config.go [hunk 5]
```diff
@@ -80,7 +101,17 @@ func resourceAviatrixControllerCertDomainConfigDelete(ctx context.Context, d *sc
...
+			certDomainConfig, err := client.GetCertDomain(ctx)
+			if err != nil {
+				return diag.Errorf("could not get cert domain: %v", err)
+			}
+			if certDomainConfig.CertDomain == "aviatrixnetwork.com" {
+				return nil
+			}
+		} else {
+			return diag.Errorf("could not reset cert domain: %v", err)
+		}
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_cert_domain_config.go]: Same comments as above

**Review by CyrusJavan — CHANGES_REQUESTED** [review]

**Review by CyrusJavan — APPROVED** [review]
Looks good to me!

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**